File: 福建省公共资源服务信息/01.js
const CryptoJs = require('./crypto-js')
const md5 = require('./md5.min.js');
r_a="B3978D054A72A7002063637CCDF6B2E5"
r_e="EB444973714E4A40876CE66BE45D5930"
r_i="B5A8904209931867"
function b(t) {
    var e = CryptoJs.enc.Utf8.parse(r_e)
      , n = CryptoJs.enc.Utf8.parse(r_i)
      , a = CryptoJs.AES.decrypt(t, e, {
        iv: n,
        mode: CryptoJs.mode.CBC,
        padding: CryptoJs.pad.Pkcs7
    });
    return a.toString(CryptoJs.enc.Utf8)
}
function s(t, e) {
    return t.toString().toUpperCase() > e.toString().toUpperCase() ? 1 : t.toString().toUpperCase() == e.toString().toUpperCase() ? 0 : -1
}
function l(t) {
    for (var e = Object.keys(t).sort(s), n = "", a = 0; a < e.length; a++)
        if (void 0 !== t[e[a]])
            if (t[e[a]] && t[e[a]]instanceof Object || t[e[a]]instanceof Array) {
                var i = JSON.stringify(t[e[a]]);
                n += e[a] + i
            } else
                n += e[a] + t[e[a]];
    return n
}
function u(e) {
    return md5.update(e)["hex"]()
}
function d(t) {
    for (var e in t)
        "" !== t[e] && void 0 !== t[e] || delete t[e];
    var n = r_a + l(t);
    return u(n)
}
// ccc={
//     "ts": 1704434378073,
//     "pageSize": 20,
//     "AREACODE": "",
//     "M_PROJECT_TYPE": "",
//     "KIND": "GCJS",
//     "GGTYPE": "",
//     "PROTYPE": "",
//     "timeType": "6",
//     "BeginTime": "2023-07-05 00:00:00",
//     "EndTime": "2024-01-05 23:59:59"
// }
// console.log(d(ccc))
// e="B3978D054A72A7002063637CCDF6B2E5BeginTime2023-07-05 00:00:00EndTime2024-01-05 23:59:59KINDGCJSpageSize20timeType6ts1704434142789"
// console.log(u(e))

File: 福建省公共资源服务信息/01.py
import json
import time
from datetime import datetime
import subprocess
from functools import partial

subprocess.Popen = partial(subprocess.Popen, encoding='utf-8')
import execjs
import requests

cookies = {
    'ASP.NET_SessionId': 'xewkqh3ex2dd5dk0hnqaz0b0',
}
with open("./01.js", 'r', encoding='utf-8') as f:
    js_data = f.read()
    JS_run = execjs.compile(js_data)
# 获取当前时间的时间戳
timestamp = datetime.now().timestamp()

# 将时间戳转换为13位格式
timestamp_13 = int(timestamp * 1000)
# print(timestamp_13)
json_data = {
    "ts": timestamp_13,
    "pageSize": 20,
    "AREACODE": "",
    "M_PROJECT_TYPE": "",
    "KIND": "GCJS",
    "GGTYPE": "",
    "PROTYPE": "",
    "timeType": "6",
    "BeginTime": "2023-07-05 00:00:00",
    "EndTime": "2024-01-05 23:59:59"
}
portalSign = JS_run.call("d", json_data)
print(portalSign)
headers = {
    'Accept': 'application/json, text/plain, */*',
    'Accept-Language': 'zh-CN,zh;q=0.9,en;q=0.8',
    'Connection': 'keep-alive',
    'Content-Type': 'application/json;charset=UTF-8',
    # 'Cookie': 'ASP.NET_SessionId=xewkqh3ex2dd5dk0hnqaz0b0',
    'DNT': '1',
    'Origin': 'https://ggzyfw.fj.gov.cn',
    'Referer': 'https://ggzyfw.fj.gov.cn/business/list/',
    'Sec-Fetch-Dest': 'empty',
    'Sec-Fetch-Mode': 'cors',
    'Sec-Fetch-Site': 'same-origin',
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36',
    'portal-sign': portalSign,
    'sec-ch-ua': '"Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"',
    'sec-ch-ua-mobile': '?0',
    'sec-ch-ua-platform': '"Windows"',
}
response = requests.post('https://ggzyfw.fj.gov.cn/FwPortalApi/Trade/TradeInfo', cookies=cookies, headers=headers,
                         json=json_data)
resp = json.loads(response.text)
t = resp['Data']
if t is not None:
    a_list = JS_run.call("b", t)
    j_list = json.loads(a_list)
    print(j_list['Table'])
    for item in j_list['Table']:
        print(item)
else:
    print(resp)
